### PR TITLE
Lucile reusable user change

### PR DESCRIPTION
**Pull request by LucileTech**
###Description
After a concerning bug, this PR has been made in order to change only a part of the User Profile without the use of putUserProfile function
In this PR this new code is used for isRehireable and Bio Status on the report page. 

###Related PR
To test this frontend PR you need to checkout the #411 backend PR.
https://github.com/OneCommunityGlobal/HGNRest/pull/411

###Mainly changes explained:
1 - Add the reusable updateUserProfileProperty 

###Before 

https://github.com/OneCommunityGlobal/HGNRest/assets/111132148/78cd8ce5-fd98-4d84-8cb3-f738ae8ad811

###After 

https://github.com/OneCommunityGlobal/HGNRest/assets/111132148/6b44e983-7a0a-4c74-a86e-90b51f4b615b

###How to test:
Git checkout Lucile-reusable-user-change in the backend
do "npm pull" and "npm install" and "npm run build" and "npm start" to run this PR locally

Git checkout Lucile-reusable-user-change in the frontend
do "npm install" and "npm run start:local"

Admin or Owner login → Reports → Reports → People → Choose someone with an end date 
You should be able to check and checkout the rehireable check box and the bio status on the right.
The chosen check should remain even if the page is refreshed.

Admin or Owner login → Reports → Reports → People → Choose someone without an end date 
You should not see the rehireable checkbox

### CODE REVIEW

File: src/components/Reports/PeopleReport/PeopleReport.jsx
Comment (suigunfjc) [inline src/components/Reports/PeopleReport/PeopleReport.jsx:508]: I have a concern regarding the current userId retrieval method, as we have made changes in this area. While I believe your proposed method is sound, I would like to confirm with @juneluo to ensure that changing this line is acceptable.

**Review by suigunfjc — COMMENTED** [review]
Hi @LucileTech , I tested your code and it works as intended, you fixed the bug on the bio status and isrehirable, GREAT JOB!:
Screenshot 2023-06-18 164530
Screenshot 2023-06-18 164613
Screenshot 2023-06-18 164633

Though I have one concern and left a comment on line 508 of PeopleReport. I will approve the PR after @juneluo review and approve on that line.

File: src/components/Reports/PeopleReport/PeopleReport.jsx
Comment (juneluo) [reply src/components/Reports/PeopleReport/PeopleReport.jsx:508]: The change looks good to me because previously `this.state.userId` was set in `componentDidMount` and using the same `userId` that `this.props.userProfile` uses as the param of the `getUserProfile`. So I think this change will work as intended.

**Comment by juneluo** [comment]
Test result: The end date information remains correct after modifying isRehireable or BioStatus value.
The code works as intended and the bug of end date has been fixed. Good job!

**Comment by one-community** [comment]
Thank you both! Should I merge this without further review or let it go through the usual process?

**Review by suigunfjc — APPROVED** [review]
Approved! Please address the comments on BE and we can merge them.

**Review by eduhorta — APPROVED** [review]
Hey, @LucileTech!

The feature is working as intended! Great job! 👍

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/b6925b90-dc49-403e-8153-d90c852d2cf7

File: src/actions/userProfile.js
Comment (cone-7) [inline src/actions/userProfile.js:65]: I think await is not necessary here

**Review by cvtqx — APPROVED** [review]
The changes work as described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/20b4f1c2-364c-4e9d-aa0a-394ebd16d34f

**Review by jonathancarpena — APPROVED** [review]
Good job! @LucileTech 👍 Works exactly like you said. Someone without an end date does not have a rehireable checkbox and the rehireable checkbox is available to those that have an end date. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/63978806/eb92c53c-ccb5-4f40-99c6-9a45bbcb34cb

**Review by mahamakhan — COMMENTED** [review]
Function works as mentioned. 

image
image

**Pull request merged**
